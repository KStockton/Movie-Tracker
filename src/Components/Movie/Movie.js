import { NavLink } from "react-router-dom";
import React from "react";
import { library } from '@fortawesome/fontawesome-svg-core'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faArrowLeft } from '@fortawesome/free-solid-svg-icons'
import { Card } from "../../Containers/Card/Card";
import PropTypes from 'prop-types'

library.add(faArrowLeft)

export const Movie = props => {

  const {
    overview,
    backdrop_path,
    poster_path,
    release_date,
    title,
    vote_average
  } = props.location.state.movie;
  
  const releaseDate = release_date.substring(0, 4);
  const rating = vote_average * 10;
  const posterUrl = `https://image.tmdb.org/t/p/w500${poster_path}`;
  const backDropUrl = `https://image.tmdb.org/t/p/w1280${backdrop_path}`;
  
  return (
    <article className="movie-card">
      <div className="movie-card-background">
        <img alt="backdrop" src={backDropUrl} />
      </div>
      <article className="movie-card-container">
        <section className="movie-card-info">
          <div>
            <h4 className="movie-text">{title}</h4>
            <br />
            <div className="movie-rating">
              <span className="vote-average"> Rating : {rating} %</span>
              <span className="release-date"> {releaseDate}</span>
            </div>
          </div>
          <br />
          <span>Description: </span>
          <br />
          <p className="movie-text movie-overview">{overview}</p>
        </section>
        <section className="movie-poster-path">
          <img alt="poster" src={posterUrl} />
        </section>
      </article>
      <NavLink className="go-back-btn" to="/">
     <FontAwesomeIcon icon="arrow-left" />
      </NavLink>
    </article>
  );
};

Card.propTypes = {
  movie: PropTypes.object
}
