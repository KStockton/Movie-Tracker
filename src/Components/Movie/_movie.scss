.movie-card {
  width: 100%;
  height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-content: center;
  align-items: center;
}

.movie-card-background {
  width: 100%;
  height: 100%;
  position: absolute;
  background-color: black;
  z-index: -1;
  & img {
    height: 100%;
    width: 100%;
    object-fit: cover;
    opacity: 0.15;
  }
}

.movie-card-container {
  position: relative;
  padding: 2em;
  z-index: 1;
  display: flex;
  height: 50%;
  width: 50em;

  & img {
    height: 100%;
  }
}
.movie-poster-path {
    width: 60%;
}
.movie-card-info {
  width: 50%;
  text-align: left;
  padding: 2em 4em;
  color: rgb(235, 235, 235);
  display: flex;
  flex-direction: column;
  & .movie-overview {
      font-size: 0.9em;
      line-height: 1.5em;
      color: rgb(202, 202, 202);
  }
  & h4 {
    font-size: 1.5em;
  }
  & div .vote-average{
      color:#47D369;
      margin-right: 0.5em;
  }
  
}

.movie-rating {
    padding: 0.6em 0;

}

.release-date {
    border: 1px solid white;
    padding: 0.3em;
}

.go-back-btn {
    text-decoration:none;
    color: white;
    background: #E50615;
    width: 5em;
    height: 2em;
    display: flex;
    justify-content: center;
    align-items: center;
    transition: 0.5s color;
    transition: 0.5s background;

}
.go-back-btn:hover {
    color: #E50615;
    background-color: white;
}