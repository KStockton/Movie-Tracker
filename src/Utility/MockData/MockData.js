
export const uncleanMovies =
[
  {
      "vote_count": 2045,
      "id": 19404,
      "video": false,
      "vote_average": 9,
      "title": "Dilwale Dulhania Le Jayenge",
      "poster_path": "/uC6TTUhPpQCmgldGyYveKRAu8JN.jpg",
      "original_language": "hi",
      "original_title": "दिलवाले दुल्हनिया ले जायेंगे",
      "genre_ids": [
          35,
          18,
          10749
      ],
      "backdrop_path": "/nl79FQ8xWZkhL3rDr1v2RFFR6J0.jpg",
      "adult": false,
      "overview": "Raj is a rich, carefree, happy-go-lucky second generation NRI. Simran is the daughter of Chaudhary Baldev Singh, who in spite of being an NRI is very strict about adherence to Indian values. Simran has left for India to be married to her childhood fiancé. Raj leaves for India with a mission at his hands, to claim his lady love under the noses of her whole family. Thus begins a saga.",
      "release_date": "1995-10-20"
  },
  {
      "vote_count": 12962,
      "id": 278,
      "video": false,
      "vote_average": 8.7,
      "title": "The Shawshank Redemption",
      "poster_path": "/9O7gLzmreU0nGkIB6K3BsJbzvNv.jpg",
      "original_language": "en",
      "original_title": "The Shawshank Redemption",
      "genre_ids": [
          18,
          80
      ],
      "backdrop_path": "/j9XKiZrVeViAixVRzCta7h1VU9W.jpg",
      "adult": false,
      "overview": "Framed in the 1940s for the double murder of his wife and her lover, upstanding banker Andy Dufresne begins a new life at the Shawshank prison, where he puts his accounting skills to work for an amoral warden. During his long stretch in prison, Dufresne comes to be admired by the other inmates -- including an older prisoner named Red -- for his integrity and unquenchable sense of hope.",
      "release_date": "1994-09-23"
  }]

  export const cleanMovies = 
  [{
     id: 19404,
     title: "Dilwale Dulhania Le Jayenge",
     backdrop_path: "/nl79FQ8xWZkhL3rDr1v2RFFR6J0.jpg",
     overview: "Raj is a rich, carefree, happy-go-lucky second generation NRI. Simran is the daughter of Chaudhary Baldev Singh, who in spite of being an NRI is very strict about adherence to Indian values. Simran has left for India to be married to her childhood fiancé. Raj leaves for India with a mission at his hands, to claim his lady love under the noses of her whole family. Thus begins a saga.",
     poster_path: "/uC6TTUhPpQCmgldGyYveKRAu8JN.jpg",
     vote_average: 9,
     release_date: "1995-10-20",
     favorited: false
  },
    {
      id: 278,
      title:  "The Shawshank Redemption",
      backdrop_path: "/j9XKiZrVeViAixVRzCta7h1VU9W.jpg",
      overview: "Framed in the 1940s for the double murder of his wife and her lover, upstanding banker Andy Dufresne begins a new life at the Shawshank prison, where he puts his accounting skills to work for an amoral warden. During his long stretch in prison, Dufresne comes to be admired by the other inmates -- including an older prisoner named Red -- for his integrity and unquenchable sense of hope.",
      poster_path: "/9O7gLzmreU0nGkIB6K3BsJbzvNv.jpg",
      vote_average: 8.7,
      release_date: "1994-09-23",
      favorited: false
  }]

  export const movieProp = {
    state: {
      movie: {
        id: 2341,
        title: "HellBoy",
        backdrop_path: "/5BkSkNtfrnT",
        favorited: false,
        overview: "walk on earth",
        poster_path: "/bk8LyaMqUtaQ9hUShuvFznQYQKR.jpg",
        release_date: "2019-04-12"
      }
    }
  }
