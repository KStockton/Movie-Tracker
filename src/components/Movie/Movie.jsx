import React from 'react';


export const Movie = (props) => {
  console.log('movieprops', props)
  return (
    <div>
      <h1>Hi</h1>
    </div>
  )
}

