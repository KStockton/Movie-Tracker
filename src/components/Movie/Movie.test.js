import React from 'react';
import {Movie} from './Movie';
import { shallow , mount} from 'enzyme';
import * as MD from '../../Utility/MockData/MockData'


const mockMovieProp = MD.movieProp

describe('Movie', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount (
        <Movie movie={mockMovieProp}/>
    )
  })

  it.skip('movie component should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
