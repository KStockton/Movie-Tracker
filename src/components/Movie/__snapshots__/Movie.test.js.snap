// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie movie component should match the snapshot 1`] = `undefined`;
